File: be-nodejs/src/libs/mappers/user.mapper.ts
import { User } from "@/models/user.model";
import { hmacSha256 } from "../hashing";
import { UserProfileResponse } from "@/types/res/user.res";

export const mapUserToUserProfileResponse = async (user: User) => {
    return {
        email: user.email,
        fullname: user.fullname,
        phone: user.phone,
        dateOfBirth: user.dateOfBirth,
        role: await hmacSha256(user.role),
        isEnable: user.isEnable,
        lastLoginDate: user.lastLoginDate,
        createdDate: user.createdDate,
        updatedDate: user.updatedDate,
    } as UserProfileResponse;
}

File: servers/MasterServices/Utils/UserMapper.cs
using MasterServices.Models;
using MasterServices.Web.Responses;
using MasterServices.Utils;

namespace MasterServices.Utils
{
    public static class UserMapper
    {
        public static UserProfileResponse MapUserToUserProfileResponse(User user, IConfiguration configuration)
        {
            return new UserProfileResponse
            {
                Email = user.Email,
                Fullname = user.Fullname,
                Phone = user.Phone,
                DateOfBirth = user.DateOfBirth,
                Role = HashingUtil.HashString(user.Role.ToString(), configuration),
                IsEnable = user.IsEnable,
                LastLoginDate = user.LastLoginDate,
                CreatedDate = user.CreatedDate,
                UpdatedDate = user.UpdatedDate
            };
        }
    }
}
